source/features/remove-label-button.tsx

```diff
@@ -0,0 +1,86 @@
+import './remove-label-button.css';
+import React from 'dom-chef';
+import XIcon from 'octicon/x.svg';
+import select from 'select-dom';
+import oneTime from 'onetime';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+import {observe} from 'selector-observer';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+
+import {getRepoURL, getConversationNumber} from '../github-helpers';
+
+const canNotEditLabels = oneTime((): boolean => !select.exists('.sidebar-labels .octicon-gear'));
+
+function updateSidebar() {
+	select('#partial-discussion-sidebar')!.dispatchEvent(new CustomEvent('socket:message', {
+		detail: {
+			name: '',
+			data: {},
+			cached: false
+		}
+	}));
+}
+
+async function removeLabelButtonClickHandler(event: delegate.Event<MouseEvent, HTMLButtonElement>): Promise<void> {
+	event.preventDefault();
+
+	const removeLabelButton = event.delegateTarget;
+
+	removeLabelButton.disabled = true;
+	await api.v3(`repos/${getRepoURL()}/issues/${getConversationNumber()!}/labels/${removeLabelButton.dataset.name!}`, {
+		method: 'DELETE'
+	});
+
+	updateSidebar();
+}
+
+function makeRemoveLabelButton(labelName: string, color: string, backgroundColor: string) {
+	const removeLabelButton = (
+		<button
+			type="button"
+			aria-label="Remove this label"
+			className="btn-link tooltipped tooltipped-nw rgh-remove-label-button"
+			data-name={labelName}
+		>
+			<XIcon/>
+		</button>
+	);
+
+	removeLabelButton.style.setProperty('--rgh-remove-label-button-bg-color', backgroundColor);
+	removeLabelButton.style.setProperty('--rgh-remove-label-button-color', color);
+
+	return removeLabelButton;
+}
+
+async function init(): Promise<void> {
+	await api.expectToken();
+
+	observe('.labels > a', {
```

Comment:
I think this will need a `:not(.rgh-something)` because we need to ensure it's not applied when the user pressed the browser’s back button.